platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/merkle/AbstractHashListener.java

```diff
@@ -156,6 +158,8 @@ public void onHashingCompleted() {
             leaves = null;
         }
         if (!finalNodesToFlush.isEmpty() || !finalLeavesToFlush.isEmpty()) {
+            assert !flushInProgress.get() : "Flush must not be in progress when hashing is complete";
+            flushInProgress.set(true);
             flush(finalNodesToFlush, finalLeavesToFlush);
```

Comment:
This should probably use compareAndSet or similar. It seems a bit fragile to just set the flag unconditionally.